### DISCUSSION TITLE
Auto format on text mode

### DISCUSSION THREAD

**Discussion by pmnzt** [josdejong/svelte-jsoneditor#400] [Q&A] [answered]
passing a json object for content.text, the output is unformatted json, then the option to format appears because of askToFormat option being true.

But i want to right away set a formatted json into the editor.

I use text mode.

and i pass the object as is to content.text, without JSON.stringify

**Comment by josdejong** [comment, marked as answer]
If you want to just format your JSON, you can do so with a one liner, see https://github.com/josdejong/svelte-jsoneditor/issues/3#issuecomment-803282031. Is that what you mean?

**Comment by pmnzt** [comment]
> If you want to just format your JSON, you can do so with a one liner, see [#3 (comment)](https://github.com/josdejong/svelte-jsoneditor/issues/3#issuecomment-803282031). Is that what you mean?

Thank you very much for the quick response! <3 

Yes ``JSON.stringify(JSON.parse(MyJsonObject), null, 2) }`` did the trick.

Though i was trying JSON.stringify(MyJsonObject, null, 2) and i was getting an invalid json text, thinking that i provided a json object when i have actually parsed it to a string already.

**Reply by josdejong** [reply]
Good to hear you have this working now :)

Have a nice weekend
